src/data/types.rs

```diff
@@ -386,3 +392,37 @@ impl FunctionType {
         *self.return_type != Type::Void
     }
 }
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use proptest::prelude::*;
+
+    use super::Type;
+
+    pub(crate) fn arb_type() -> impl Strategy<Value = Type> {
+        let leaf = prop_oneof![
+            Just(Type::Void),
+            Just(Type::Bool),
+            any::<bool>().prop_map(Type::Char),
+            any::<bool>().prop_map(Type::Short),
+            any::<bool>().prop_map(Type::Int),
+            any::<bool>().prop_map(Type::Long),
+            Just(Type::Float),
+            Just(Type::Double),
+            //Type::Enum(Option<InternedStr>, Vec<(InternedStr, i64)>),
```

Comment:
I thought you implemented arbitrary for InternedStr? Is there something else making this tricky?

Reply:
No, there is a function providing an arbitrary one, but not a direct trait implementation, as that is kind of tricky. If you want to give it a go though, have at it! :D